snapcraft.yaml

```diff
@@ -0,0 +1,52 @@
+# Copyright The ORAS Authors.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+
+# you may not use this file except in compliance with the License.
+
+# You may obtain a copy of the License at
+
+#
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+#
+
+# Unless required by applicable law or agreed to in writing, software
+
+# distributed under the License is distributed on an "AS IS" BASIS,
+
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+# See the License for the specific language governing permissions and
+
+# limitations under the License.
+name: oras
+version: git
+summary: Package for the ORAS CLI tool
+description: |
+  A snap package for the ORAS CLI tool, which is used for managing OCI artifacts.
+base: core22
+confinement: strict
+grade: devel
+
+parts:
+  oras:
+    plugin: go
+    source: .
+    build-snaps:
+      - go/1.20/stable
+    build-packages:
+      - make
+    stage-packages:
+      - ca-certificates
+    override-build: |
+      make "build-linux-$CRAFT_TARGET_ARCH"
```

Comment:
Got two stupid questions on this:
1) looks like this will use `build-linux-*` targets provided by oras, if we are going to add a new snap package for ppc64el, do we need to add a new target named `build-linux-ppc64el` right? Is there any good practice for it?

2) About cross-platform building, I am able to build amd64 package on my amd64 host, how can I customize to build arm64 package?

Reply:
Regarding your first question, it is necessary to add a new target. As for the best practice, I don't have a definitive answer.

For your second question, I will need to investigate further.

Reply:
we can also put directly the go command to avoid creating new target

Reply:
for your second question we can do something like this

```
architectures:
  - build-on: amd64
    build-for: all
```

Reply:
> For your second question, I will need to investigate further.

Well I am just playing, I think in the end we will use https://github.com/snapcore/action-build to do cross-platform building

> we can also put directly the go command to avoid creating new target

Is it possible to skip the building and publish based on github packages like `curl -LO "https://github.com/oras-project/oras/releases/download/v${VERSION}/oras_${VERSION}_${CRAFT_TARGET_ARCH}.tar.gz`

Reply:
or 

```
architectures:
  - amd64
  - arm64
```

Reply:
@qweeah We can use curl, yes, but you will need to change the version field on the snapcraft. yaml each time you release a new one

Reply:
or get the last version with some git command